tfe/data_source_organization_membership.go

```diff
@@ -40,28 +40,48 @@ func dataSourceTFEOrganizationMembershipRead(d *schema.ResourceData, meta interf
 	// Create an options struct.
 	options := &tfe.OrganizationMembershipListOptions{
 		Include: []tfe.OrgMembershipIncludeOpt{tfe.OrgMembershipUser},
+		Emails:  []string{email},
 	}
 
-	for {
-		organizationMembershipsList, err := tfeClient.OrganizationMemberships.List(ctx, organization, options)
-		if err != nil {
-			return fmt.Errorf("Error retrieving organization memberships: %v", err)
-		}
+	oml, err := tfeClient.OrganizationMemberships.List(ctx, organization, options)
+	if err != nil {
+		return fmt.Errorf("Error retrieving organization memberships: %v", err)
+	}
 
-		for _, organizationMembership := range organizationMembershipsList.Items {
-			if organizationMembership.User.Email == email {
-				d.SetId(organizationMembership.ID)
-				return resourceTFEOrganizationMembershipRead(d, meta)
-			}
+	switch len(oml.Items) {
+	case 0:
+		return fmt.Errorf("Could not find organization membership for organization %s and email %s", organization, email)
+	case 1:
+		if oml.Items[0].User.Email != email {
```

Comment:
Just one suggestion... this case (and the team case) has become odd but you and I both know it's totally reasonable. Maybe just comment about how this and the default case are to support versions of TFE that do not support the email filter option?